### PR TITLE
fix: fix the crd and defaulter

### PR DESCRIPTION
**Pull request by zhiying-lin**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
- fix the comment
- fix the crd
- add the crd validation

Fixes #

**Requirements**:

- [ ] run `make reviewable` for basic local test
- [ ] Read and followed fleet-networking's [Code of conduct](https://github.com/Azure/fleet-networking/blob/main/CODE_OF_CONDUCT.md).
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

File: pkg/common/defaulter/trafficmanagerprofile_test.go [hunk 1]
```diff
@@ -60,6 +60,57 @@ func TestSetTrafficManagerProfile(t *testing.T) {
 				},
 			},
 		},
+		{
```
Comment (jwtty) [inline pkg/common/defaulter/trafficmanagerprofile_test.go:63]: Add a test case for 30 IntervalInSeconds and nil TimeInSeconds?

Comment (jwtty) [reply pkg/common/defaulter/trafficmanagerprofile_test.go:63]: oh I think the 1st case covers this one. nvm

**Review by jwtty — APPROVED** [review]
